src/read_ic.c

```diff
@@ -1,1342 +1,1398 @@
+                   && blocknr != IO_HSML
+#endif
+#ifdef EOS_CARRIES_TEMPERATURE
+                   && blocknr != IO_EOSTEMP
+#endif
+#ifdef EOS_CARRIES_ABAR
+                   && blocknr != IO_EOSABAR
+#endif
+#ifdef EOS_CARRIES_YE
+                   && blocknr != IO_EOSYE
+#endif
+#ifdef EOS_TILLOTSON
+                   && blocknr != IO_EOSCOMP
+#endif
+#if defined(HYDRO_MESHLESS_FINITE_VOLUME) && ((HYDRO_FIX_MESH_MOTION==1)||(HYDRO_FIX_MESH_MOTION==2)||(HYDRO_FIX_MESH_MOTION==3))
+                   && blocknr != IO_PARTVEL
+#endif
+#if defined(BH_GRAVCAPTURE_FIXEDSINKRADIUS)
+                   && blocknr != IO_SINKRAD
+#endif
+#if defined(CHIMES) && !defined(CHIMES_INITIALISE_IN_EQM)
+                   && blocknr != IO_CHIMES_ABUNDANCES
+#endif
+#ifdef PIC_MHD
+                   && blocknr != IO_GRAINTYPE
+#endif
+                   )
+#if defined(GDE_DISTORTIONTENSOR) && defined(GDE_READIC)
+                    if(RestartFlag == 0 && (blocknr > IO_U && blocknr != IO_SHEET_ORIENTATION))
+                        if(RestartFlag == 0 && (blocknr > IO_U && blocknr != IO_INIT_DENSITY))
+                            if(RestartFlag == 0 && (blocknr > IO_U && blocknr != IO_CAUSTIC_COUNTER))
+#endif
+                                continue;	/* ignore all other blocks in initial conditions */
+
+
+            if(RestartFlag == 0 && (blocknr == IO_GENERATION_ID || blocknr == IO_CHILD_ID)) {continue;}
+#if defined(NO_CHILD_IDS_IN_ICS) || defined(ASSIGN_NEW_IDS)
+            if(blocknr == IO_GENERATION_ID || blocknr == IO_CHILD_ID) {continue;}
+#endif
+            if((RestartFlag == 0) && (All.InitGasTemp > 0) && (blocknr == IO_U)) {continue;}
+
+
+#ifdef MHD_B_SET_IN_PARAMS
+            if(RestartFlag == 0 && blocknr == IO_BFLD) {continue;}
+#endif
+
+#ifdef SUBFIND
+            if(RestartFlag == 2 && blocknr == IO_HSMS) {continue;}
+#endif
+
+#ifdef AGS_HSML_CALCULATION_IS_ACTIVE
+#ifndef AGS_OUTPUTGRAVSOFT
+            if(blocknr == IO_AGS_SOFT) {continue;}
+#endif
+#ifndef AGS_OUTPUTZETA
+            if(blocknr == IO_AGS_ZETA) {continue;}
+#endif
+#endif
+            
+#ifdef COSMIC_RAYS_EVOLVE_SPECTRUM_SPECIAL_SNAPSHOTRESTART
+            if(RestartFlag == 2 && blocknr == IO_COSMICRAY_SLOPES) {continue;}
+#if (COSMIC_RAYS_EVOLVE_SPECTRUM_SPECIAL_SNAPSHOTRESTART==2)
+            if(RestartFlag == 2 && blocknr == IO_COSMICRAY_ENERGY) {continue;}
+#endif
+#endif
+
+#if !defined(RADTRANSFER)
+            if(RestartFlag == 2 && blocknr == IO_RADGAMMA) {continue;}
+#endif
+
+#if defined(IO_MOLECFRAC_NOT_IN_ICFILE)
+            if(RestartFlag == 2 && blocknr == IO_MOLECULARFRACTION) {continue;}
+#endif
+
+            
+            if(blocknr == IO_HSMS) {continue;}
+
+#ifdef TURB_DIFF_DYNAMIC
+            if (RestartFlag == 0 && blocknr == IO_TURB_DYNAMIC_COEFF) {continue;}
+#endif
+
+            if(ThisTask == readTask)
+            {
+                get_dataset_name(blocknr, buf);
+                printf("reading block %d (%s)...\n", bnr, buf);

```

Comment:
Once the slug variables are not skipped, this code should execute and when the code is reading in the initial conditions, we should see a message like "reading block IO_SLUG_STATE_RNG..." printed to the simulation log file.